Investigate AssertionError in randomSplit

```
] - should support randomSplit() *** FAILED *** (3 seconds, 711 milliseconds)
[info]   java.lang.AssertionError: apply@{Matchers.scala:6770}:888/ParMultiDo(anon$7).output: 
[info] Expected: <500L>
[info]      but: was <600L>
[info]   at org.apache.beam.sdk.testing.PAssert$PAssertionSite.capture(PAssert.java:169)
[info]   at org.apache.beam.sdk.testing.PAssert.thatSingleton(PAssert.java:449)
[info]   at org.apache.beam.sdk.testing.PAssert.thatSingleton(PAssert.java:441)
[info]   at com.spotify.scio.testing.SCollectionMatchers$$anon$5$$anon$6$$anonfun$apply$3.apply(SCollectionMatchers.scala:215)
[info]   at com.spotify.scio.testing.SCollectionMatchers$$anon$5$$anon$6$$anonfun$apply$3.apply(SCollectionMatchers.scala:214)
[info]   at com.spotify.scio.testing.SCollectionMatchers.com$spotify$scio$testing$SCollectionMatchers$$m(SCollectionMatchers.scala:86)
[info]   at com.spotify.scio.testing.SCollectionMatchers$$anon$5$$anon$6.apply(SCollectionMatchers.scala:217)
[info]   at com.spotify.scio.testing.SCollectionMatchers$$anon$5$$anon$6.apply(SCollectionMatchers.scala:211)
[info]   at com.spotify.scio.testing.SCollectionMatchers$SingleMatcher.apply(SCollectionMatchers.scala:65)
[info]   at com.spotify.scio.testing.SCollectionMatchers$SingleMatcher.apply$(SCollectionMatchers.scala:65)
[info]   ...
[info]   Cause: java.lang.AssertionError: Expected: <500L>
[info]      but: was <600L>
[info]   at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
[info]   at org.junit.Assert.assertThat(Assert.java:956)
[info]   at org.junit.Assert.assertThat(Assert.java:923)
[info]   at org.apache.beam.sdk.testing.PAssert$AssertIsEqualTo.apply(PAssert.java:1448)
[info]   at org.apache.beam.sdk.testing.PAssert$AssertIsEqualTo.apply(PAssert.java:1438)
[info]   at org.apache.beam.sdk.testing.PAssert$CheckRelationAgainstExpected.apply(PAssert.java:1146)
[info]   at org.apache.beam.sdk.testing.PAssert$CheckRelationAgainstExpected.apply(PAssert.java:1126)
[info]   at org.apache.beam.sdk.testing.PAssert.doChecks(PAssert.java:1425)
[info]   at org.apache.beam.sdk.testing.PAssert$SingletonCheckerDoFn.processElement(PAssert.java:1418)
[info]   at org.apache.beam.sdk.testing.PAssert$SingletonCheckerDoFn$DoFnInvoker.invokeProcessElement(Unknown Source)
[info]   ...
```

--- a/scio-test/src/test/scala/com/spotify/scio/values/SCollectionTest.scala
+++ b/scio-test/src/test/scala/com/spotify/scio/values/SCollectionTest.scala
@@ -358,24 +358,25 @@ class SCollectionTest extends PipelineSpec {
 
   it should "support randomSplit()" in {
     runWithContext { sc =>
-      def round(c: Long): Long = math.round(c / 100.0) * 100
-      val p1 = sc.parallelize(0 to 1000).randomSplit(Array(0.3, 0.7))
-      val p2 = sc.parallelize(0 to 1000).randomSplit(Array(0.2, 0.3, 0.5))
+      val total = 5000
+      def round(c: Long): Double = math.round(c * 10.0 / total) / 10.0
+      val p1 = sc.parallelize(0 to total).randomSplit(Array(0.3, 0.7))
+      val p2 = sc.parallelize(0 to total).randomSplit(Array(0.2, 0.3, 0.5))
       p1.length shouldBe 2
       p2.length shouldBe 3
-      p1(0).count.map(round) should containSingleValue(300L)
-      p1(1).count.map(round) should containSingleValue(700L)
-      p2(0).count.map(round) should containSingleValue(200L)
-      p2(1).count.map(round) should containSingleValue(300L)
-      p2(2).count.map(round) should containSingleValue(500L)
+      p1(0).count.map(round) should containSingleValue(0.3)
+      p1(1).count.map(round) should containSingleValue(0.7)
+      p2(0).count.map(round) should containSingleValue(0.2)
+      p2(1).count.map(round) should containSingleValue(0.3)
+      p2(2).count.map(round) should containSingleValue(0.5)
 
-      val (pa, pb) = sc.parallelize(0 to 1000).randomSplit(0.3)
-      val (pc, pd, pe) = sc.parallelize(0 to 1000).randomSplit(0.2, 0.3)
-      pa.count.map(round) should containSingleValue(300L)
-      pb.count.map(round) should containSingleValue(700L)
-      pc.count.map(round) should containSingleValue(200L)
-      pd.count.map(round) should containSingleValue(300L)
-      pe.count.map(round) should containSingleValue(500L)
+      val (pa, pb) = sc.parallelize(0 to 5000).randomSplit(0.3)
+      val (pc, pd, pe) = sc.parallelize(0 to 5000).randomSplit(0.2, 0.3)
+      pa.count.map(round) should containSingleValue(0.3)
+      pb.count.map(round) should containSingleValue(0.7)
+      pc.count.map(round) should containSingleValue(0.2)
+      pd.count.map(round) should containSingleValue(0.3)
+      pe.count.map(round) should containSingleValue(0.5)
     }
   }
 

